pkg/gen/node-templates/provider.ts.mustache

```diff
@@ -1,9 +1,110 @@
 // *** WARNING: this file was generated by the Pulumi Kubernetes codegen tool. ***
 // *** Do not edit by hand unless you're certain you know what you are doing! ***
 
+import * as fs from "fs";
+
 import * as pulumi from "@pulumi/pulumi";
 import * as inputApi from "./types/input";
 import * as outputApi from "./types/output";
+import * as yaml from "js-yaml";
+import * as glob from "glob";
+
+export interface YamlOpts {
+    files?: string[] | string;
+    yaml?: string[] | string;
+    transformations?: ((o: any) => void)[];
+}
+
+/**
+ * Yaml creates a set of Kubernetes resources from Kubernetes YAML text. The YAML text may be
+ * supplied using any of the following `YamlOpts`:
+ *
+ *   1. Using a filename or a list of filenames:
+ *        a. `{files: "foo.yaml"}`
+ *        b. `{files: ["foo.yaml", "bar.yaml"]}`
+ *   2. Using a file pattern or a list of file patterns:
+ *        a. `{files: "*.yaml"}`
+ *        b. `{files: ["foo/*.yaml", "bar/*.yaml"]}`
+ *   3. Using a literal string containing YAML, or a list of such strings:
+ *        a. `{yaml: "(LITERAL YAML HERE)"}`
+ *        b. `{yaml: ["(LITERAL YAML HERE)", "(MORE YAML)"]}`
+ *   4. Any combination of files, patterns, or YAML strings:
+ *        a. `{files: "foo.yaml", yaml: "(LITERAL YAML HERE)"}`
+ */
+export class Yaml {
+    public readonly resources: { [key: string]: pulumi.CustomResource };
+
+    constructor(config: YamlOpts, opts?: pulumi.CustomResourceOptions) {
+        const yamlTexts = [];
+
+        if (config.files !== undefined) {
+            let files: string[] = [];
+            if (Object.prototype.toString.call(config.files) === "[object String]") {
+                files = glob.sync(<string>config.files);
+            } else {
+                for (const file of config.files) {
+                    files.push(...glob.sync(file));
+                }
+            }
+
+            for (const file of files) {
+                yamlTexts.push(fs.readFileSync(file).toString());
+            }
+        }
+
+        if (config.yaml !== undefined) {
+            if (Object.prototype.toString.call(config.files) === "[object String]") {
```

Comment:
Same here

Reply:
Was this meant to be a check on `config.yaml`?